README.rst

```diff
@@ -58,12 +58,164 @@ Python:
+
+Timing
+------
+You can use software timing or hardware timing to control when a signal is generated. With hardware timing,
+a digital signal, such as a clock on your device, controls the rate of generation. With software timing,
+the rate at which the samples are generated is determined by the software and operating system instead of
+by the measurement device. A hardware clock can run much faster than a software loop. A hardware clock is
+also more accurate than a software loop.
+Refer to the `Timing, Hardware Versus Software <https://www.ni.com/docs/en-US/bundle/ni-daqmx/page/hardwresoftwretiming.html>`_ for more information.
+
+Example code to acquire finite amount of data using hardware timing:
+
+.. code-block:: python
+
+  >>> import nidaqmx
+  >>> from nidaqmx.constants import AcquisitionType, READ_ALL_AVAILABLE
+  >>> with nidaqmx.Task() as task:
+  ...     task.ai_channels.add_ai_voltage_chan("Dev1/ai0")
+  ...     task.timing.cfg_samp_clk_timing(1000.0, sample_mode=AcquisitionType.FINITE, samps_per_chan=10)
+  ...     data = task.read(READ_ALL_AVAILABLE)
+  ...     print("Acquired data: [" + ", ".join(f"{value:f}" for value in data) + "]")
+  ...
+  AIChannel(name=Dev1/ai0)
+  Acquired data: [-0.149693, 2.869503, 4.520249, 4.704886, 2.875912, -0.006104, -2.895596, -4.493698, -4.515671, -2.776574]
+
+TDMS Logging
+------------
+Technical Data Management Streaming (TDMS) is a binary file format that allows for high-speed data logging.
+When you enable TDMS data logging, NI-DAQmx can stream data directly from the device buffer to the hard disk.
+Refer to the `TDMS Logging <https://www.ni.com/docs/en-US/bundle/ni-daqmx/page/datalogging.html>`_ for more information.
+
+Example code to acquire finite amount of data and log it to a TDMS file:
+
+.. code-block:: python
+
+  >>> import nidaqmx
+  >>> from nidaqmx.constants import AcquisitionType, LoggingMode, LoggingOperation, READ_ALL_AVAILABLE
+  >>> with nidaqmx.Task() as task:
+  ...     task.ai_channels.add_ai_voltage_chan("Dev1/ai0")
+  ...     task.timing.cfg_samp_clk_timing(1000.0, sample_mode=AcquisitionType.FINITE, samps_per_chan=10)
+  ...     task._in_stream.configure_logging("TestData.tdms", LoggingMode.LOG_AND_READ, operation=LoggingOperation.CREATE_OR_REPLACE)
+  ...     data = task.read(READ_ALL_AVAILABLE)
+  ...     print("Acquired data: [" + ", ".join(f"{value:f}" for value in data) + "]")
+  ...
+  AIChannel(name=Dev1/ai0)
+  Acquired data: [-0.149693, 2.869503, 4.520249, 4.704886, 2.875912, -0.006104, -2.895596, -4.493698, -4.515671, -2.776574]
+
+Plot Data
+---------
+You can use the 3rd-party module **matplotlib.pyplot** to plot a waveform for the acquired data.
+For more information on how to use **matplotlib.pyplot** module, refer to the
+`Pyplot tutorial <https://matplotlib.org/stable/tutorials/pyplot.html#sphx-glr-tutorials-pyplot-py>`_.
+
+Example code to plot waveform for acquired data using **matplotlib.pyplot** module:
```

Comment:
Are you adding any matplotlib example programs to the examples directory? It seems weird to demonstrate it in the readme but not in an example program.

Reply:
Sure, I have added the matplotlib example in another PR.